### PR TITLE
sign and only show verified public chat messages

### PR DESCRIPTION
**Pull request by neuroscr**
Address oxen-io/session-desktop-temp#1065

### CODE REVIEW

**Comment by sachaaaaa** [comment]
I think travis failed because of linting. Try `yarn lint` twice?

File: js/modules/loki_public_chat_api.js
Comment (sachaaaaa) [inline js/modules/loki_public_chat_api.js]: You can use spread syntax to copy arrays and objects:
```js
const arrayCopy = [...myArray];
const objectCopy = { ... myObject};
```

File: js/modules/loki_public_chat_api.js
Comment (sachaaaaa) [inline js/modules/loki_public_chat_api.js]: Does this work with passing a string?
I've always thought it would only accept an ArrayBuffer

File: js/modules/loki_public_chat_api.js
Comment (sachaaaaa) [inline js/modules/loki_public_chat_api.js]: The whole snippet above can go if you move `sig` and `sigver` up.

File: js/modules/loki_public_chat_api.js
Comment (sachaaaaa) [inline js/modules/loki_public_chat_api.js]: Move the scope of `sig` and `sigver` to be declared next to `sigValid`

File: js/modules/loki_public_chat_api.js
Comment (sachaaaaa) [inline js/modules/loki_public_chat_api.js]: initialise to false?

File: js/modules/loki_public_chat_api.js
Comment (sachaaaaa) [inline js/modules/loki_public_chat_api.js]:
```suggestion
      payload.annotations[0].value.quote = quote;
```

File: js/modules/loki_public_chat_api.js
Comment (sachaaaaa) [inline js/modules/loki_public_chat_api.js]: Feels like wrapping this whole additional signature checking in a function might help with readability:
```js
// ... existing code 
if (!verifySignature(adnMessage)) {
  return;
}
// ... existing code
```

File: js/modules/loki_public_chat_api.js
Comment (neuroscr) [reply js/modules/loki_public_chat_api.js]: yes, I think I tried an arrayBuffer and had problems

File: js/modules/loki_public_chat_api.js
Comment (neuroscr) [reply js/modules/loki_public_chat_api.js]: not following

File: js/modules/loki_public_chat_api.js
Comment (neuroscr) [reply js/modules/loki_public_chat_api.js]: ugh good catch 🤦‍♂

File: js/modules/loki_public_chat_api.js
Comment (sachaaaaa) [inline js/modules/loki_public_chat_api.js]: Actually, since you want to copy but only take some of the fields, you can do it in one go:
```js
const annoCopy = _.omit(adnMessage.annotations, ['value.sig', 'value.sigver']);
```

File: js/modules/loki_public_chat_api.js
Comment (sachaaaaa) [inline js/modules/loki_public_chat_api.js]:
```
ObjToSign => objToSign
             ^
```

File: js/modules/loki_public_chat_api.js
Comment (sachaaaaa) [inline js/modules/loki_public_chat_api.js]: Use `ObjToSign.version` ?

File: js/modules/loki_public_chat_api.js
Comment (sachaaaaa) [reply js/modules/loki_public_chat_api.js]: All good then :)

File: js/modules/loki_public_chat_api.js
Comment (neuroscr) [reply js/modules/loki_public_chat_api.js]: it's not the version of the `network.loki.messenger.publicChat` annotation, it's the version of the signature format is my thinking. You could infer the signature version from the annotation version I suppose, but this current way gives up a bit more freedom and precision.

File: js/modules/loki_public_chat_api.js
Comment (neuroscr) [reply js/modules/loki_public_chat_api.js]: refactored it a bit different, I think it made everything cleaner/simpler

File: js/modules/loki_public_chat_api.js
Comment (neuroscr) [reply js/modules/loki_public_chat_api.js]: resolved

File: js/modules/loki_public_chat_api.js
Comment (neuroscr) [reply js/modules/loki_public_chat_api.js]: resolved

File: js/modules/loki_public_chat_api.js
Comment (sachaaaaa) [reply js/modules/loki_public_chat_api.js]: Sorry if my question wasn't clear.
Basically there are two locations where we set a version to `1`, and to me it looks like it's the same value being repeated:
On the receiving/verification end, we use `annotation[0].value.sigver` to create `verifyObj.version`.
So I assumed that here on the sending end,
```js
const ObjToSign = {
      version: 1,
      ...
```
and
```js
payload.annotations[0].value.sigver = 1;
```
should be the same value too? If it's the case, let's put that into a constant global variable `SIGNATURE_VERSION` or something :)

File: js/modules/loki_public_chat_api.js
Comment (sachaaaaa) [inline js/modules/loki_public_chat_api.js]:
```suggestion
        const messengerData = await this.getMessengerData(adnMessage);
```

File: js/modules/loki_public_chat_api.js
Comment (neuroscr) [reply js/modules/loki_public_chat_api.js]: resolved

File: js/modules/loki_public_chat_api.js
Comment (neuroscr) [reply js/modules/loki_public_chat_api.js]: ah makes sense

Comment (neuroscr) [reply js/modules/loki_public_chat_api.js]: I think `objToSign.version` is the right move here. The version is closely tied to the code path, we may need to be able to write to multiple versions in the future, so a singular constant at the top doesn't make much sense to me.

**Review by sachaaaaa — APPROVED** [review]
Looks good to me!
Waiting for CI before merging..

**Review by Mikunj — APPROVED** [review]

**Pull request merged**
